routers/api/v1/repo/pull.go

```diff
@@ -671,8 +671,8 @@ func parseCompareInfo(ctx *context.APIContext, form api.CreatePullRequestOption)
 		ctx.ServerError("GetUserRepoPermission", err)
 		return nil, nil, nil, nil, "", ""
 	}
-	if !perm.CanReadIssuesOrPulls(true) {
-		log.Trace("ParseCompareInfo[%d]: cannot create/read pull requests", baseRepo.ID)
+	if !perm.CanReadIssuesOrPulls(true) && !perm.CanWrite(models.UnitTypeCode) {
```

Comment:
It seems it's not reasonable.

Reply:
I don't understand. Are you saying that this is wrong?

What should the permissions test be?

Reply:
For a `parseCompare` we should have two permissions, one is for base repo, another is to head repo.

Reply:
OK So I'm not sure that's what's currently being tested.